src/emitters/emitter.sqns.ts

```diff
@@ -317,6 +319,29 @@ export class SqnsEmitter implements IEmitter {
     return true;
   }
 
+  getPayloadToEmit(eventName: string, options?: IEmitOptions, ...args: any[]): EmitPayload {
+    const topic: Topic = {
+      name: eventName,
+      isFifo: !!options?.isFifo,
+      exchangeType: options?.exchangeType || ExchangeType.Fanout,
+      separateConsumerGroup: options?.consumerGroup,
+    };
+    const queueOrTopic =
+      topic.exchangeType === ExchangeType.Queue
+        ? this.getQueueUrl(this.getQueueName(topic))
+        : this.getTopicArn(this.getTopicName(topic));
+    const message: IMessage = {
+      messageGroupId: options?.partitionKey || topic.name,
+      eventName: topic.name,
+      data: args,
+      messageAttributes: options?.MessageAttributes,
+      deduplicationId: options?.deduplicationId,
+    };
+    return topic.exchangeType === ExchangeType.Queue ? this.sqsProducer.constructQueueMessageRequest(queueOrTopic, message, {
+        delay: options?.delay || DEFAULT_MESSAGE_DELAY,
+      }) : this.snsProducer.constructPublishResponse(queueOrTopic, message);
+  }
```

Comment:
It's better to use if/else for readability

Reply:
Modified as Expected.